servers/quarkus-server/src/main/java/org/projectnessie/server/providers/RocksDbSource.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright (C) 2022 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.server.providers;
+
+import javax.enterprise.inject.Disposes;
+import javax.enterprise.inject.Produces;
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import org.projectnessie.server.config.VersionStoreConfig;
+import org.projectnessie.versioned.persist.rocks.ImmutableRocksDbConfig;
+import org.projectnessie.versioned.persist.rocks.RocksDbInstance;
+
+@Singleton
+public class RocksDbSource {
+  @Inject VersionStoreConfig.RocksVersionStoreConfig rocksConfig;
+
+  @Produces
+  @Singleton
+  RocksDbInstance createRocksDbInstance() {
+    RocksDbInstance rocksDbInstance = new RocksDbInstance();
+    rocksDbInstance.configure(
+        ImmutableRocksDbConfig.builder().dbPath(rocksConfig.getDbPath()).build());
+    rocksDbInstance.initialize();
+    return rocksDbInstance;
+  }
+
+  void close(@Disposes RocksDbInstance rocksDbInstance) {
+    rocksDbInstance.close();
```

Comment:
maybe add a null check in case it never actually got initialized

Reply:
How shall that happen? That instance is inhjected here by the container.